drv/gimlet-hf-server/src/main.rs

```diff
@@ -108,7 +116,16 @@ fn main() -> ! {
     // trouble. Someday we will need more flexability here.
     let log2_capacity = {
         let mut idbuf = [0; 20];
-        qspi.read_id(&mut idbuf);
+        let result = qspi.read_id(&mut idbuf);
+        if result.is_err() {
+            // If we can't read the id there's a good chance nothing else is going to
+            // work. `panic` would probably just be a crash loop. looping forever
+            // is slightly more polite.
+            loop {
+                // We are dead now.
+                hl::sleep_for(1000);
+            }
```

Comment:
elsewhere, we have tasks that die permanently by calling `sys_recv_notification` with an empty notification mask, like this: https://github.com/oxidecomputer/hubris/blob/7a24b884ba1907c1d45432760ad189d1a606e4a2/drv/gimlet-seq-server/src/main.rs#L174-L184

i think this is a bit nicer as we don't get periodically woken up by the timer, remember that we are just looping on it, and go back to sleep. not that this case is particularly worth optimizing for, though.

Reply:
IIRC Humility shows tasks in `sys_recv_notification(0)` as `(DEAD)`, so there's precedent for treating it as a canonically dead state.